src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_device_connection_status_request.cc

```diff
@@ -0,0 +1,220 @@
+
+GetDeviceConnectionStatusRequest::~GetDeviceConnectionStatusRequest() {}
+
+void GetDeviceConnectionStatusRequest::Run() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  smart_objects::SmartObject& object = *message_;
+  smart_objects::SmartObject& msg_params_so = object[strings::msg_params];
+
+  if (msg_params_so.keyExists(strings::device) &&
+      !msg_params_so[strings::device].empty()) {
+    smart_objects::SmartObject& devices = msg_params_so[strings::device];
+    DeviceDataRetriever device_data_retriever(devices, application_manager_);
+    for (size_t i = 0; i < devices.length(); ++i) {
+      smart_objects::SmartObject& current_device = devices[i];
+
+      std::string device_id;
+      if (!GetMandatoryParameters(current_device, device_id)) {
+        LOG4CXX_WARN(logger_,
+                     "Invalid data for device: " << device_id << " found");
+        SendErrorResponse(
+            correlation_id(),
+            static_cast<hmi_apis::FunctionID::eType>(function_id()),
+            hmi_apis::Common_Result::INVALID_DATA,
+            "");
+        return;
+      }
+
+      device_data_retriever.FillDataForDeviceAtIndex(device_id, i);
+    }
+  } else {
+    const std::vector<std::string> devices_ids =
+        application_manager_.GetPolicyHandler().GetDevicesIDs();
+    smart_objects::SmartObject& devices = msg_params_so[strings::device];
+    devices = smart_objects::SmartObject(smart_objects::SmartType_Array);
+    DeviceDataRetriever device_data_retriever(devices, application_manager_);
+    std::for_each(
+        devices_ids.begin(), devices_ids.end(), device_data_retriever);
+  }
+
+  object[strings::params][hmi_response::code] =
+      hmi_apis::Common_Result::SUCCESS;
+  application_manager_.GetRPCService().ManageHMICommand(message_);
+}
+
+bool GetDeviceConnectionStatusRequest::GetMandatoryParameters(
+    const smart_objects::SmartObject& device_data,
+    std::string& out_device_id) const {
+  LOG4CXX_AUTO_TRACE(logger_);
+  std::string device_name;
+  if (!(GetKeyStringValue(device_data, strings::id, out_device_id) &&
+        GetKeyStringValue(device_data, strings::name, device_name))) {
+    LOG4CXX_WARN(logger_, "Invalid data for key value(s)");
+    return false;
+  }
+  return true;
+}
+
+bool GetDeviceConnectionStatusRequest::GetKeyStringValue(
+    const smart_objects::SmartObject& device_data,
+    const std::string& key,
+    std::string& out_value) const {
+  LOG4CXX_AUTO_TRACE(logger_);
+  // APPLINK-28265 - Invalidate HMI request with missing parameter
+  if (!device_data.keyExists(key)) {
+    LOG4CXX_WARN(logger_, "Invalid data: Key " << key << " does not exist.");
+    return false;
+  }
+  out_value = device_data[key].asString();
+  // APPLINK-28279 - Invalidate HMI request with at least one empty String
+  // parameter
+  if (out_value.empty()) {
+    LOG4CXX_WARN(logger_, "Invalid data: Key " << key << " empty.");
+    return false;
+  }
+  if (IsSyntaxWrong(out_value)) {
```

Comment:
@ZhdanovP what about case when string contains not only invalid characters? I don't think that string like `aaa\n\taaa` is correct

Reply:
@AKalinich-Luxoft yes this is incorrect one, but we are checking all characters in the srting by InvalidCharacterChecker for reject such cases